tests/periph_pm/main.c

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (C) 2016-2018 Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Power management peripheral test.
+ *
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "periph/pm.h"
+#ifdef MODULE_PM_LAYERED
+#ifdef MODULE_PERIPH_RTC
+#include "periph/rtc.h"
+#endif
+#include "pm_layered.h"
+#endif
+#include "shell.h"
+
+#ifdef MODULE_PM_LAYERED
+static int parse_mode(int argc, char **argv)
+{
+    if (argc < 2) {
```

Comment:
This should be moved to each command function handler and if a wrong number of arguments is given, the command prints a usage message `Usage: unblock <power mode>` or `Usage: unblock_rtc <power mode> <duration>`.
Otherwise the error message doesn't give any hint on how to correctly use the command. Maybe introduce a `_check_usage(int argc, int expected_argc)` function for this ?

Then the parse_mode and parse_duration could be simplified with this signature:
```c
static int _parse_mode(const char * mode)
static int _parse_duration(const char * duration)
```